WalletWasabi/Crypto/Transcript.cs

```diff
@@ -0,0 +1,115 @@
+using NBitcoin.Secp256k1;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using WalletWasabi.Crypto.Groups;
+using WalletWasabi.Crypto.Randomness;
+using WalletWasabi.Crypto.StrobeProtocol;
+
+namespace WalletWasabi.Crypto.ZeroKnowledge
+{
+	public delegate IEnumerable<Scalar> NoncesSequence();
```

Comment:
fwiw in my WIP branch i've renamed everything to do with nonces so it's always "secret nonce" and "public nonce", i think that's a bit clearer even though you could argue that the type (`Scalar` vs. `GroupElement`) conveys the same information